### PR TITLE
fix: init and startup issues

### PR DESCRIPTION
**Pull request by mdebarros**
- fixes for missing dependency dns-packet (updated dockerfile ordering)
- fixed issue with config.js when env var was missing
- fixed audit-resolve
- chore: added automated ci-cd release
    - updated circleci config for automated releases
    - fixed caching issue with npm dependencies
    - updated readme with automated release information
    - standardized node version via nvmrc and in circleci config

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const defaultConf = {
         tls: {
             host: process.env.PATHFINDER_TLS_HOST,
             port: process.env.PATHFINDER_TLS_PORT,
-            rejectUnauthorized: process.env.PATHFINDER_REJECT_UNAUTHORIZED.toLowerCase() !== 'false'
+            rejectUnauthorized: process.env.PATHFINDER_REJECT_UNAUTHORIZED && process.env.PATHFINDER_REJECT_UNAUTHORIZED.toLowerCase() !== 'false'
```
Comment (partiallyordered) [inline src/config.js:28]: Same as this, I think:
```suggestion
            rejectUnauthorized: process.env.PATHFINDER_REJECT_UNAUTHORIZED?.toLowerCase() !== 'false'
```

**Comment by partiallyordered** [comment]
It looks like `dns-packet` is present in the internal pathfinder lib dependency manifest. I don't think it shouldn't be necessary/present in the top-level manifest, should it?

As an aside, the organisation of this repo might have caused more pain than it's worth, even though it scopes dependencies better. It could be worth reorganising this repo, or perhaps moving to yarn, as you mentioned it might have better support for internal packages.

**Comment by mdebarros** [comment]
> It looks like `dns-packet` is present in the internal pathfinder lib dependency manifest. I don't think it shouldn't be necessary/present in the top-level manifest, should it?

Fresh install (i.e. removed node_modules) does not install the dependency. Also if you are referencing a dependency directly in "our" code, it should be added a dependency in our package regardless if another dependency pulls it in.

The reason for that is because we want to control the version that we use in our code.

**Comment by partiallyordered** [comment]
Is this not the only usage of that lib?
https://github.com/mojaloop/als-oracle-pathfinder/blob/5e854eed6afcaa12c720dd51a813c808e7aef0db/src/lib/pathfinder/tlsResolver.js#L24
The dependency specified here?
https://github.com/mojaloop/als-oracle-pathfinder/blob/5e854eed6afcaa12c720dd51a813c808e7aef0db/src/lib/pathfinder/package.json#L19

What am I missing?

File: src/config.js (hunk 1 above)
Comment (partiallyordered) [reply src/config.js:28]: Requires newer node version, out of scope for this PR

File: Dockerfile [hunk 2]
```diff
@@ -8,6 +8,7 @@ RUN apk add --no-cache -t build-dependencies git make gcc g++ python libtool aut
     && npm install -g node-gyp
 
 COPY ./package.json ./package-lock.json ./init-account-lookup.sql ./init-central-ledger.sql /opt/als-oracle-pathfinder/
+COPY ./src /opt/als-oracle-pathfinder/src
```
Comment (partiallyordered) [inline Dockerfile:11]: The advantage of this is that lower layers in the image change less frequently, which is better for storage (pretty unimportant) and build speed (can be important for development turn-around)
```suggestion
COPY ./src/lib/pathfinder/package.json ./src/lib/pathfinder/package-lock.json /opt/als-oracle-pathfinder/src/lib/pathfinder/
```

File: Dockerfile [hunk 3]
```diff
@@ -8,6 +8,7 @@ RUN apk add --no-cache -t build-dependencies git make gcc g++ python libtool aut
...
 
 RUN npm install --production
```
Comment (partiallyordered) [inline Dockerfile:13]: May as well update this while we're making changes in this area.
```suggestion
RUN npm ci --production
```

File: Dockerfile (hunk 2 above)
Comment (partiallyordered) [reply Dockerfile:11]: If you accept this, you'll need to revert the deletion of this line later in the file.

Comment (mdebarros) [reply Dockerfile:11]: I would prefer to stick to what is currently there as that is the way "other" mojaloop images are built using this pattern.

**Review by partiallyordered — APPROVED** [review]
I've basically ignored the CI additions in this PR; assuming they're pretty standard across ML repos.

**Pull request merged**